src/generator_p.h

```diff
@@ -31,22 +33,77 @@ std::string generateDoubleCode(const std::string &value);
  *
  * The private implementation for the Generator class.
  */
-struct Generator::GeneratorImpl
+struct Generator::GeneratorImpl: public Logger::LoggerImpl
 {
-    AnalyserModelPtr mModel;
-
     std::string mCode;
 
     GeneratorProfilePtr mProfile = GeneratorProfile::create();
 
+    std::map<AnalyserModelPtr, std::map<AnalyserVariablePtr, bool>> mTrackedVariables;
+
     void reset();
 
-    bool modelHasOdes() const;
-    bool modelHasNlas() const;
+    std::string analyserVariableIndexString(const AnalyserModelPtr &analyserModel,
+                                            const AnalyserVariablePtr &analyserVariable);
+
+    bool doIsTrackedEquation(const AnalyserEquationPtr &analyserEquation, bool tracked);
+
+    bool isTrackedEquation(const AnalyserEquationPtr &analyserEquation);
+    bool isUntrackedEquation(const AnalyserEquationPtr &analyserEquation);
+
+    bool doIsTrackedVariable(const AnalyserModelPtr &analyserModel, const AnalyserVariablePtr &analyserVariable,
+                             bool tracked = true);
+    bool doIsTrackedVariable(const AnalyserVariablePtr &analyserVariable, bool tracked);
```

Comment:
Not a virtual function and parameter overloading will differentiate these methods.